### PR TITLE
Adapt execution policy name changes from C++17

### LINKED ISSUE #2390: Adapt execution policy name changes from C++17
While the Parallelsim TS has been now adopted tobe part of C++17, there have been a couple of name changes which require corresponding changes in HPX:

 - [x] `par_vec` has been renamed to `par_unseq` (see #2436)
 - [x] execution policies now live in `namespace std::execution`  (see #2436)
 - [x] execution policies now don't have `'execution'` in their name anymore (i.e. the former  `std::sequential_execution_policy` is now named `std::execution::sequential_policy`)  (see #2436)
 - [x] the header `<execution_policy>` is now named `<execution>`  (see #2436)
 - [x] `inner_product` was renamed to `transform_reduce` (see #2437)
 - [x] various parameter sequence changes have been adopted (for `transform_reduce` etc.) (see #2437)

Other changes (we need to decide whether we will adopt those):

 ~- [ ] any exception thrown from any of the algorithms now calls `std::terminate`~
 ~- [ ] the algorithm `adjacent_difference` has been removed~

### PR DESCRIPTION
**Pull request by hkaiser**
This fixes #2390

### CODE REVIEW

File: hpx/parallel/datapar/execution_policy.hpp [hunk 1]
```diff
@@ -24,18 +24,18 @@
 #include <type_traits>
 #include <utility>
 
-namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1)
+namespace hpx { namespace parallel { namespace execution
```
Comment (sithhell) [inline hpx/parallel/datapar/execution_policy.hpp]: why is this one not versioned anymore?

File: CMakeLists.txt [hunk 2]
```diff
@@ -796,6 +796,14 @@ if(HPX_WITH_BOOST_CHRONO_COMPATIBILITY)
   hpx_add_config_define(HPX_HAVE_BOOST_CHRONO_COMPATIBILITY)
 endif()
 
+# HPX_WITH_EXECUTION_POLICY_COMPATIBILITY: introduced in V1.0.0
+hpx_option(HPX_WITH_EXECUTION_POLICY_COMPATIBILITY BOOL
+    "Enable old execution policy names in API (default: OFF)"
+    OFF ADVANCED)
```
Comment (sithhell) [inline CMakeLists.txt]: Can we have this on by default?

Comment (hkaiser) [inline CMakeLists.txt]: Sure. Good point.

File: docs/manual/build_system/cmake_variables.qbk [hunk 3]
```diff
@@ -78,6 +79,7 @@ The options are split into these categories:
         [[[#build_system.cmake_variables.HPX_WITH_DATAPAR_BOOST_SIMD] `HPX_WITH_DATAPAR_BOOST_SIMD:BOOL`][Enable data parallel algorithm support using the external Boost.SIMD library (default: OFF)]]
         [[[#build_system.cmake_variables.HPX_WITH_DATAPAR_LIBFLATARRAY] `HPX_WITH_DATAPAR_LIBFLATARRAY:BOOL`][Enable data parallel algorithm support using the external LibFlatArray (default: OFF)]]
         [[[#build_system.cmake_variables.HPX_WITH_DATAPAR_VC] `HPX_WITH_DATAPAR_VC:BOOL`][Enable data parallel algorithm support using the external Vc library (default: OFF)]]
+        [[[#build_system.cmake_variables.HPX_WITH_EXECUTION_POLICY_COMPATIBILITY] `HPX_WITH_EXECUTION_POLICY_COMPATIBILITY:BOOL`][Enable old execution policy names in API (default: OFF)]]
```
Comment (sithhell) [inline docs/manual/build_system/cmake_variables.qbk]: the new default should be reflected here

File: hpx/parallel/datapar/execution_policy.hpp (hunk 1 above)
Comment (sithhell) [inline hpx/parallel/datapar/execution_policy.hpp]: Ok. I clearly miss where the versioning is done here. I miss the inline namespace v2 or something similar.

Comment (hkaiser) [inline hpx/parallel/datapar/execution_policy.hpp]: Right, my misunderstanding - I mixed that up with the executor work I'm doing right now. Please disregard my first comment.

I assume, you are talking about moving (keeping) the `datapar` execution policies into a versioned namespace. I agree, that should probably be done.

File: docs/manual/build_system/cmake_variables.qbk (hunk 3 above)
Comment (hkaiser) [inline docs/manual/build_system/cmake_variables.qbk]: RIght.

Comment (hkaiser) [inline docs/manual/build_system/cmake_variables.qbk]: BTW, I think we should remove this file from the repository altogether as it is regenerated by cmake.

**Comment by hkaiser** [comment]
Can I go ahead and merge this now?

**Comment by hkaiser** [comment]
The comments left by @sithhell have been addressed.

**Pull request merged**